cmd/asset-syncer/sync.go

```diff
@@ -92,23 +92,37 @@ var syncCmd = &cobra.Command{
 		}
 
 		// Check if the repo has been already processed
-		if manager.RepoAlreadyProcessed(models.Repo{Namespace: repo.Namespace, Name: repo.Name}, checksum) {
+		lastChecksum := manager.LastChecksum(models.Repo{Namespace: repo.Namespace, Name: repo.Name})
+		logrus.Infof("Last checksum: %v", lastChecksum)
+		if lastChecksum == checksum {
 			logrus.WithFields(logrus.Fields{"url": repo.URL}).Info("Skipping repository since there are no updates")
 			return
 		}
 
-		charts, err := repoIface.Charts()
-		if err != nil {
-			logrus.Fatal(err)
-		}
+		// First filter the list of charts (still without applying custom filters)
+		repoIface.FilterIndex()
 
-		if err = manager.Sync(models.Repo{Name: repo.Name, Namespace: repo.Namespace}, charts); err != nil {
-			logrus.Fatalf("Can't add chart repository to database: %v", err)
+		shallowSyncs := []bool{false}
+		if lastChecksum == "" {
+			// If the repo has never been processed, run first a shallow sync to give early feedback
+			// then sync all the repositories
+			shallowSyncs = []bool{true, false}
 		}
 
-		// Fetch and store chart icons
-		fImporter := fileImporter{manager, netClient}
-		fImporter.fetchFiles(charts, repoIface)
+		for _, shallow := range shallowSyncs {
+			charts, err := repoIface.Charts(shallow)
```

Comment:
For cases like this where the argument name isn't used so it's not obvious what `repoIface.Charts(shallow)` does, I'd re-work so that it reads as `repoIface.Charts(fetch_latest_only)`, for example.